### PR TITLE
Add connect only natscontext package

### PR DESCRIPTION
**Pull request by ripienaar**
This adds a version of natscontext that supports only the parts needed to connect to nats and none of the management features the CLI needs, this is in order to keep the usage API and dependencies low

### CODE REVIEW

File: natscontext/context.go [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+	UserJwt             string   `json:"user_jwt"`
+	ColorScheme         string   `json:"color_scheme"`
+	TLSFirst            bool     `json:"tls_first"`
+	WinCertStoreType    string   `json:"windows_cert_store"`
+	WinCertStoreMatchBy string   `json:"windows_cert_match_by"`
+	WinCertStoreMatch   string   `json:"windows_cert_match"`
+	WinCertStoreCaMatch []string `json:"windows_ca_certs_match"`
+}
+
+// Connect connects to the NATS server configured by the named context, empty name connects to selected context. If the name is a absolute name to a context configuration that will be used
```
Comment (piotrpio) [inline natscontext/context.go:70]: Probably also worth to document that the `Settings` returned here are the selected context's parameters.

File: natscontext/context.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+func readCtxFromFile(file string) string {
+	parent, err := parentDir()
+	if err != nil {
+		return ""
+	}
+
+	currentFile := filepath.Join(parent, "nats", file)
+
+	_, err = os.Stat(currentFile)
+	if os.IsNotExist(err) {
```
Comment (piotrpio) [inline natscontext/context.go:205]: What about other errors here? Shouldn't we return "" on any error from `Stat`?

File: natscontext/context.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+	fc, err := os.ReadFile(currentFile)
+	if err != nil {
+		return ""
+	}
+
+	return strings.TrimSpace(string(fc))
+}
+
+func selectedContext() string {
+	return readCtxFromFile("context.txt")
```
Comment (piotrpio) [inline natscontext/context.go:218]: I had no idea we had this default, is it documented anywhere? If not, we probably should :)

**Review by piotrpio — COMMENTED** [review]
Looks good, just a few nits.

File: natscontext/context.go (hunk 3 above)
Comment (ripienaar) [reply natscontext/context.go:218]: Thanks will get to the other comments - for this, the ADR fully specifies it https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-21.md

Comment (ripienaar) [reply natscontext/context.go:218]: `nats ctx select` creates this file :)

File: natscontext/context.go (hunk 2 above)
Comment (ripienaar) [reply natscontext/context.go:205]: Could indeed - I am not sure though will the read sometimes pass and stat not maybe due to weird selinux constraints? So I give the read below a chance to succeed even if the stat failed with non missing file errors.  If read also fail then its `""`

File: natscontext/context.go [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+	nopts, err := c.nATSOptions(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return nats.Connect(c.config.URL, nopts...)
+}
+
+// nATSOptions creates NATS client configuration based on the contents of the context
+func (c *context) nATSOptions(opts ...nats.Option) ([]nats.Option, error) {
```
[the remark this answers is not in the document]
Comment (ripienaar) [reply natscontext/context.go]: good call, in jsm.go its NATSOptions so I just did the quickest unexport :) Sorted now

File: natscontext/context.go (hunk 1 above)
Comment (ripienaar) [reply natscontext/context.go]: done this

File: natscontext/context.go (hunk 3 above)
Comment (piotrpio) [reply natscontext/context.go:219]: Thanks!

File: natscontext/context.go (hunk 2 above)
Comment (piotrpio) [reply natscontext/context.go:206]: True, it'll fail anyway if there is stat error, so this will suffice.

**Review by piotrpio — APPROVED** [review]
LGTM!

**Pull request merged**